fluent/test/langneg_test.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+import assert from 'assert';
+import { negotiateLanguages } from '../src/langneg';
+
+describe('Language Negotiation', function() {
+  it('exact match', function() {
+    assert.deepEqual(
+      negotiateLanguages(['en-US'], ['en-US']),
+      ['en-US']);
+
+    assert.deepEqual(
+      negotiateLanguages(['en'], ['en-US', 'en']),
+      ['en']);
+
+    assert.deepEqual(
+      negotiateLanguages(['en-US'], ['en', 'en-US']),
+      ['en-US']);
+
+    assert.deepEqual(
+      negotiateLanguages(['ja-JP'], ['ja-JP-mac', 'ja', 'ja-JP']),
+      ['ja-JP']);
+
+    assert.deepEqual(
+      negotiateLanguages(['ja-JP-mac'], ['ja-JP-mac', 'ja', 'ja-JP']),
+      ['ja-JP-mac']);
+
+    assert.deepEqual(
+      negotiateLanguages(['en-US'], ['pl', 'en-US']),
+      ['en-US']);
+
+    assert.deepEqual(
+      negotiateLanguages(['pl'], ['pl', 'en-US']),
+      ['pl']);
+
+    assert.deepEqual(
+      negotiateLanguages(['fr', 'pl', 'de-DE'], ['pl', 'en-US', 'de-DE']),
+      ['pl', 'de-DE']);
+  });
+
+  it('parent match', function() {
+    assert.deepEqual(
+      negotiateLanguages(['en-US'], ['en-GB', 'en']),
+      ['en']);
+
+    assert.deepEqual(
+      negotiateLanguages(['de-AT'], ['de']),
+      ['de']);
+
+    assert.deepEqual(
+      negotiateLanguages(['fr-CA', 'de-DE'], ['fr', 'it', 'de']),
+      ['fr', 'de']);
+  });
+
+  it('likely subtag', function() {
+    assert.deepEqual(
+      negotiateLanguages(['en'], ['en-US', 'de']),
+      ['en-US']);
+
+    assert.deepEqual(
+      negotiateLanguages(['en'], ['en-GB', 'de']),
+      ['en-GB']);
+
+    assert.deepEqual(
+      negotiateLanguages(['en'], ['en-GB', 'de', 'en-US']),
+      ['en-US']);
+
+    assert.deepEqual(
+      negotiateLanguages(['fr'], ['el', 'de', 'fr-FR']),
+      ['fr-FR']);
+
+    assert.deepEqual(
+      negotiateLanguages(['ja'], ['ja-JP']),
+      ['ja-JP']);
+
+    assert.deepEqual(
+      negotiateLanguages(['ja-JP'], ['ja-JP-mac', 'ja']),
+      ['ja']);
+
+    assert.deepEqual(
+      negotiateLanguages(['ja-JP-mac'], ['ja-JP', 'ja']),
+      ['ja-JP']);
+
+    assert.deepEqual(
+      negotiateLanguages(['ja-JP-mac'], ['ja']),
+      ['ja']);
+
+    assert.deepEqual(
+      negotiateLanguages(['fr'], ['fr-CA', 'fr-FR']),
+      ['fr-FR']);
+
+    assert.deepEqual(
+      negotiateLanguages(['az-IR'], ['az-Latn', 'az-Arab']),
+      ['az-Arab']);
+
+    assert.deepEqual(
+      negotiateLanguages(['en-Latn-GB', 'en-Latn-IN'], ['en-IN', 'en-GB']),
+      ['en-GB', 'en-IN']);
+
+    assert.deepEqual(
+      negotiateLanguages(['sr-RU'], ['sr-Cyrl', 'sr-Latn']),
+      ['sr-Latn']);
+
+    assert.deepEqual(
+      negotiateLanguages(['sr'], ['sr-Cyrl', 'sr-Latn']),
+      ['sr-Cyrl']);
```

Comment:
I spent half a day trying to find out why cldr says these things about Serbian, and to the best of my findings, this is just made up. Notably, sr-Latn-RS is given a writingPopulation of 5%,  because

    <reference type="R1017">For languages not customarily written, the writing population is artificially set to 5% in the absence of better information.</reference>